app/src/org/commcare/views/widgets/RecordingFragment.java

```diff
@@ -428,6 +428,19 @@ private void registerAudioRecordingConfigurationChangeCallback() {
             @Override
             public void onRecordingConfigChanged(List<AudioRecordingConfiguration> configs) {
                 super.onRecordingConfigChanged(configs);
+                if (recorder == null) {
+                    return;
+                }
+
+                if (hasRecordingGoneSilent(configs)) {
+                    if (!inPausedState) {
+                        pauseRecording();
+                    }
+                } else {
+                    if (inPausedState) {
+                        resumeRecording();

```

Comment:
why should we ever resume a silent recording ?

Reply:
If we navigate back to CommCare, bringing it to the top and depending on the configuration of the other voice recording app, this might trigger a change in the recording configuration and CommCare will start capturing sound again. In this situation, we want to resume the recording without any action from the user.

Reply:
I tend to disagree with this, If we resume recording without any user intervention it would seem to user that recording was never paused. Insted when user comes back they should see a message that recording was paused due to user navigating away and give user option to resume recording.

Reply:
I understand. The rationale here was that because the user didn't directly action to pause the recording, CC should resume it in the same terms. But given that we will provide guidance to the user, it makes sense to let the user resume the recording.